HungarianAlgorithm/Hungarian/ProblemInstance.cs

```diff
@@ -47,5 +47,43 @@ public void Write(TextWriter? textWriter = null)
                 textWriter.WriteLine($"{i + 1} {x} {y}");
             }
         }
+
+        public decimal[,] CreateDistancesDecimalMatrix()
+        {
+            var costMatrix = new decimal[N * K, N];
```

Comment:
```suggestion
            var costMatrix = new decimal[N * K, N * K];
```

Reply:
Z tego co patrzyłem, to wszędzie mamy 'zawijanie' do N. Puściłem też kilka instancji i wydaje się, że wszystko jest w porządku. Ale czujność jest wskazana :)